src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,236 @@
+            } else {
+                indexValidation(index);
+                linkAtMiddle(new Node<>(value), index);
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findByIndex(index).data;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = findByIndex(index);
+        T returnValue = node.data;
+        node.data = value;
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index);
+        if (index == 0) {
+            return removeFirst();
+        } else if (index == size - 1) {
+            return removeLast();
+        } else {
+            return removeByIndex(index);
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        int index = 0;
+        Node<T> node = startNode;
+        while (node != null) {
+            T data = node.data;
+            if (data == object || object != null && object.equals(data)) {
+                remove(index);
+                return true;
+            }
+            index++;
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private T removeLast() {
+        Node<T> nodeToRemove = endNode;
+        unlink(endNode);
+        return nodeToRemove.data;
+    }
+
+    private T removeFirst() {
+        Node<T> nodeToRemove = startNode;
+        unlink(startNode);
+        return nodeToRemove.data;
+    }
+
+    private void unlink(Node<T> node) {
+        if (node.previous == null) {
+            startNode = startNode.next;
+            size--;
+        } else {
+            endNode = endNode.previous;
+            size--;
+        }
+    }
+
+    private T removeByIndex(int index) {
+        Node<T> node = findByIndex(index);
+        if (size == 2) {
+            if (index == 1) {
+                startNode.next = null;
+                endNode = null;
+            } else if (index == 0) {
+                startNode = endNode;
+                endNode = null;
+                startNode.next = null;
+            }
+        } else if (size == 1) {
+            node = startNode;
+            startNode = null;
+        } else {
+            node.previous.next = node.next;
+            node.next.previous = node.previous;
+        }
+        size--;
+        return node.data;
+    }
+
+    private void linkLast(Node<T> node) {
+        node.previous = endNode;
+        endNode.next = node;
+        endNode = node;
+        size++;
+    }
+
+    private void linkFirst(Node<T> node) {
+        startNode.previous = node;
```

Comment:
You have kind of 2 methods to set head linkFirst and initializeStart node i suggest to collaplse them (as well as linkLast and initializeEnd)